recipes/clhpp/meta.yaml

```diff
@@ -0,0 +1,76 @@
+{% set name = "clhpp" %}
+{% set version = "2.0.10" %}
+{% set sha256 = "fa27456295c3fa534ce824eb0314190a8b3ebd3ba4d93a0b1270fc65bf378f2b" %}
+
+{% set OSX_VARIANT = os.environ.get("OSX_VARIANT", "native") %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/KhronosGroup/OpenCL-CLHPP/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script_env:
+    - OSX_VARIANT
+
+{% if OSX_VARIANT != "native" %}
+  skip: True  # [not (osx and py27)]
+{% else %}
+  skip: True  # [not py27]
+{% endif %}
+
+  features:
+{% if OSX_VARIANT == "pocl" %}
+    - osx_pocl_opencl            # [osx]
+{% endif %}
+
+  always_include_files:
+    - include/CL/cl.hpp
+    - include/CL/cl2.hpp
+
+requirements:
+  build:
+    - toolchain
+    - python
+    - cmake
+    - ocl-icd                    # [linux]
+    - khronos-opencl-icd-loader  # [win]
+{% if OSX_VARIANT == "pocl" %}
+    - osx-pocl-opencl            # [osx]
+    - pocl >=0.14                # [unix]
+{% endif %}
+
+  run:
+    - khronos-opencl-icd-loader  # [win]
+    - ocl-icd                    # [linux]
+{% if OSX_VARIANT == "pocl" %}
+    - osx-pocl-opencl            # [osx]
+    - pocl >=0.14                # [unix]
+{% endif %}
```

Comment:
Even though we are just producing C++ headers, they seem to be dependent on the OpenCL implementation used. So have left these in as run time dependencies. That way we know which implementation the C++ headers should be used with.